### PR TITLE
Remove bitcoin dependency

### PR DESCRIPTION
**Pull request by aneeshvaidya**
Addresses #1200

### CODE REVIEW

File: raiden/tests/unit/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import hashlib
+
+from raiden.utils import crypto
+
+
+def test_privtopub():
+    privkey = hashlib.sha256(b'secret').hexdigest()
```
Comment (palango) [inline raiden/tests/unit/test_utils.py]: Any reason why you don't use the [`raiden.utils.sha3` function](https://github.com/raiden-network/raiden/blob/master/raiden/utils/__init__.py#L34)?

File: raiden/utils/crypto.py [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
+from coincurve import PrivateKey
 
 
 def privtopub(raw_privkey):
-    pubkey = bitcoin.privtopub(raw_privkey)
-    raw_pubkey = bitcoin.encode_pubkey(pubkey, 'bin_electrum')
-    assert len(raw_pubkey) == 64
-    return raw_pubkey
+    # either bytes or hex
+    if not isinstance(raw_privkey, bytes):
```
Comment (palango) [inline raiden/utils/crypto.py]: Do we need this check? At least [here](https://github.com/raiden-network/raiden/blob/master/raiden/tests/utils/blockchain.py#L249) the argument type is `bytes`.

File: raiden/tests/unit/test_utils.py (hunk 1 above)
Comment (aneeshvaidya) [reply raiden/tests/unit/test_utils.py]: I'll make that change, forgot about the utility function

File: raiden/utils/crypto.py (hunk 2 above)
Comment (aneeshvaidya) [reply raiden/utils/crypto.py]: If [here](https://github.com/raiden-network/raiden/blob/8c49717634e8e4303b10275c906f5fd446c374d4/raiden/accounts.py#L209) is in bytes, then most likely not.

Comment (palango) [reply raiden/utils/crypto.py]: I checked and `decode_keystore_json` returns bytes. So we can safely remove the type check and conversion.

Comment (palango) [reply raiden/utils/crypto.py]: I'd also say we can move this function to the `__init__` file.

**Comment by aneeshvaidya** [comment]
@palango should I squash these commits and rebase against master?

**Comment by LefterisJP** [comment]
@aneeshvaidya yes plz

**Comment by aneeshvaidya** [comment]
Done. I ended up finding a function in the `__init__.py` called `privatekey_to_address` that does the conversion from private to public key before calling `publickey_to_address`, so I just extracted that piece into its own function.

**Comment by palango** [comment]
Hej @aneeshvaidya 

to make sure that your function returns the right result I wrote a small script:
```py
from bitcoin import privtopub as ptp_bitcoin
from raiden.utils import privtopub as ptp_raiden
from raiden.utils import sha3

privkey = sha3(b'secret')

pub_bitcoin = ptp_bitcoin(privkey)
pub_raiden = ptp_raiden(privkey)

print(pub_bitcoin.hex())
print(pub_raiden.hex())
```

This returns the following on my system, meaning that there the function returns different results from the bitcoin one:
```
python privtopub.py
04c283b0507c4ec6903a49fac84a5aead951f3c38b2c72b69da8a70a5bac91e9c705f70c7554b26e82b90d2d1bbbaf711b10c6c8b807077f4070200a8fb4c6b771
c283b0507c4ec6903a49fac84a5aead951f3c38b2c72b69da8a70a5bac91e9c705f70c7554b26e82b90d2d1bbbaf711b10c6c8b807077f4070200a8fb4c6b771
```

I'm not sure why this is, but will investigate a bit further.

**Comment by aneeshvaidya** [comment]
The current implementation in `crypto.py` is the following

```python
def privtopub(raw_privkey):
    pubkey = bitcoin.privtopub(raw_privkey)
    raw_pubkey = bitcoin.encode_pubkey(pubkey, 'bin_electrum')
    assert len(raw_pubkey) == 64
    return raw_pubkey
```

The `encode_pubkey` function is essentially cutting that first byte. I believe this is because the public key is prefixed with a 0x04 byte.

**Comment by palango** [comment]
@aneeshvaidya I looked a bit deeper and your implementation is fine. Could you add an additional `assert` in the `privtopub` function to check that the first byte is `b'\x04'`.
```py
assert raw_pubkey.startswith(b'\x04')
```

Please rebase on top of master as well and I will merge your PR.

**Comment by aneeshvaidya** [comment]
@palango sure thing!

**Comment by palango** [comment]
Mh, another problem popped up. `ethereum` internally relies on `bitcoin`, so it displays the following warning when `bitcoin` isn't installed.

```
Failed to import bitcoin. This is not a fatal error but does
mean that you will not be able to determine the address from
your wallet file.
```

I propose to merge these changes now but not removing the bitcoin dependency for now.

**Pull request merged**